src/langcheck/metrics/custom_text_quality.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from jinja2 import Environment, Template, meta
+
+from langcheck.metrics._validation import validate_parameters_custom_evaluator
+from langcheck.metrics.eval_clients import EvalClient
+from langcheck.metrics.metric_value import MetricValue
+
+
+def custom_evaluator(generated_outputs: list[str] | str | None,
+                     prompts: list[str] | str | None,
+                     sources: list[str] | str | None,
+                     reference_outputs: list[str] | str | None,
+                     eval_model: EvalClient, metric_name: str,
+                     score_map: dict[str, float], template_path: str,
+                     language: str) -> MetricValue[float | None]:
+    '''Calculates the scores of a custom evaluator. The EvalClient will first
+    assess the provided inputs using the prompt template, and then convert those
+    assessments into scores using the score map.
```

Comment:
We need to explain what parameters are allowed in Jinjia templates here (probably want a `.md` page for that too)

Reply:
Besides the allowed parameters, we also mention some interface the EvalClient expects such as
- The LLM needs to choose one "Evaluation Result" option from the ones listed in the prompt
- These options should correspond to the keys of `score_map`
- The keys should not contain non-ascii characters

Reply:
Added this info in the function docstring! Agreed that we'd want to add a page in readthedocs too - I can add that in a follow-up PR